protocol/messenger_handler.go

```diff
@@ -1027,58 +1027,75 @@ importMessageArchivesLoop:
 			// wait for all archives to be processed first
 			downloadedArchiveID := archiveIDsToImport[0]
```

Comment:
I notice that this gets the first archiveID, which is then passed into `[]string{downloadedArchiveID}` to satisfy the function signature of `ExtractMessagesFromHistoryArchives`.

`ExtractMessagesFromHistoryArchives` is only called in one place, so I wonder if this function should expect `archiveIDs []string` and rather expect a `archiveID string`?

Reply:
This has historically changed over time indeed.
Originally `ExtractMessagesFromHistoryArchives` took a list of archive IDs and would then extract all messages for every archive matching a given ID.

This also means however that it would extract **all** messages for the given archive IDs, which means there's more messages to process/import/save. 

Because we need to wait for this operation to finish before the task can be cancelled, in case it has been cancelled, we've reduced this to only ever process one archive at a time such that, if the process is marked as cancelled, we can stop the process midflight and don't have to wait for all messages to be processed.

The only reason I haven't changed the method signature back then is because I tend to avoid breaking changes.
But you're right that we can probably change this by now.